### ISSUE TITLE
🎉 Version 2 RC3 is available.

### ISSUE THREAD

**Issue by benjamin-hull** [stevenschobert/instafeed.js#643]
This issue is to notify everyone that v2 is on its way. We're putting migration and usage instructions together now, and will have an initial version available very soon.

Stay tuned!

**Comment by benjamin-hull** [comment]
If any particularly keen folks would like to take a peek, we have some initial code, documentation and guides up for the switch to v2. This version is currently in a Release Candidate stage, and some things may change, but it would be great to hear people's initial feedback. _Please don't use this in a production setting yet, if having the Instagram feed fail would be a critical problem for you._ (Of course, we're keenly aware that the legacy version has around 24 hours to live anyway! 😬).

Please check the [Project Wiki](https://github.com/stevenschobert/instafeed.js/wiki) for the migration guide and new options.

You can see [a super-simple demo of v2 at codepen](https://codepen.io/companionstudio/pen/rNVPGOz).

**Comment by jnicol** [comment]
Oh my goodness, I'm blown away at how much work must have gone into this. I was half way through building my own token generator service but decided to hold off to see what you guys came up with. I thought you would hook into a third party service, or launch a commercial service, and had no idea you'd build a free self-hosted service for us. I just set up an instance of your Instagram Token Generator on Heroku, and it works a treat. Thank you so much.

I'll post back again once the I've tested the new version of Instafeed, to let you know how it works for me.

**Comment by benjamin-hull** [comment]
Thanks for your kind words @jnicol - it was a bit of a challenge to work out something that is a) free, and b) easy enough for novice users to deploy and c) de-centralized (mainly because we know we don't have the resources to look after a centralized version).

Heroku is probably overkill for something like the agent, but the almost-one-click deploy makes it worthwhile, I think!

**Comment by paolovador** [comment]
Hi, my client is using npm to update instafeed.js.
would it be possible for you to update https://www.npmjs.com/package/instafeed.js to list the new v2.0.0rc1 in there ?
thanks a lot for your hard work

**Comment by benjamin-hull** [comment]
Hi @paolovador - we’ll likely wait until we have a 2.0.0 release finalised before we update npm etc, and we’ve already identified some additional features that will go in (pagination, tags support), so there’ll likely be an “RC2”. We really only called it RC1 to give people the chance to switch over in the ~24 hours between our release and the original API cut off.

Did you see that the deadline to switch has been extended until the end of June? This is great because it gives us much needed time to finish the remaining features and testing for the 2.0.0 release.

**Comment by paolovador** [comment]
Yeah I saw [@Luke-SF's comment ](https://github.com/stevenschobert/instafeed.js/issues/640#issuecomment-606402923) that's great news since my client still hasn't decided whether waiting for your V2 (at the time a small possibility) or implementing ourselves the API ahah.
I'm glad you managed to do all of this.
Even though I'm surely going to integrate the instagram token refresh in the ASP.NET back-end of my client's site, I'll try to keep the front-end of instafeed.js if the stable V2 is available in time.
Thanks a lot anyway !

**Comment by celsowhite** [comment]
Great work. Thanks so much. I just tried it out and was able to successfully deploy the Instagram Token Agent app to Heroku. Simple to setup and the 'Deploy' button is especially helpful. The Instafeed api is working well too.

Appreciate the persistence in keeping the project alive and working within the new API restrictions.

**Comment by ngocthinh192** [comment]
Thanks so much for your API.
That was a huge upgrade. But I realize the API is not supporting IE browsers. Including IE 11, IE 10, IE 9, ...

**Comment by stevenschobert** [comment]
> Thanks so much for your API.
> 
> That was a huge upgrade. But I realize the API is not supporting IE browsers. Including IE 11, IE 10, IE 9, ...
> 
> 

Internet explorer **should** be supported. Tracking this issue in #651.

**Comment by LABCAT** [comment]
@benjamin-hull when do you think we can expect an official 2.0 release to be available to install via npm?

**Comment by stevenschobert** [comment]
Hey @LABCAT! The tentative plan right now is to finish RC2, which contains a few bug fixes and improvements, and release that on NPM. The RC2 release won't be the "full v2", but you should be able to install it once we have it pushed up.

**Comment by stevenschobert** [comment]
Heya @LABCAT -- just letting you know that v2.0.0-rc2 is available on NPM. Its tagged as a beta release, so you have to explicitly give the version when installing:

```
npm i instafeed.js@2.0.0-rc2
```

**Comment by NickDeckerDevs** [comment]
Not exactly sure where I would leave this.. I've never really used github to communicate with a developer -- That being said I'm trying to use 2.0 and get specific hashtags from the account, but I'm only able to pull in 25 and filter them 

```
limit: 100,
filter: function(item) {
  return item.caption.indexOf('{{ module.instagram_post_filter }}') > -1
}
```

This will only return 25 of them, and none of them match the filter... is there a better way to do this or am I SOL?

**Comment by stevenschobert** [comment]
Hey folks! Just want to let everyone know that v2.0.0-rc3 is now available here on GitHub and NPM:

* [Release on GitHub](https://github.com/stevenschobert/instafeed.js/releases/tag/v2.0.0-rc3)
* [Release on NPM](https://www.npmjs.com/package/instafeed.js/v/2.0.0-rc3)

RC3 includes 1 bug fix, ES module support, and pagination support.

**Comment by stevenschobert** [comment]
Hey @NickDeckerDevs! Sorry for taking a while to get back to you.

One thing to know about the `filter` option, is that it gets applied _after_ the images have been fetched from Instagram, which could mean you end up with less images than your `limit`. However, in v2.0.0-rc3, I've added a new `apiLimit` option that you can set to a higher value than your `limit`, to help work around that. Feel free to try it out, and open another issue if you have problems!

**Comment by NickDeckerDevs** [comment]
@stevenschobert I appreciate this, thank you.

**Comment by mylastore** [comment]
> If any particularly keen folks would like to take a peek, we have some initial code, documentation and guides up for the switch to v2. This version is currently in a Release Candidate stage, and some things may change, but it would be great to hear people's initial feedback. _Please don't use this in a production setting yet, if having the Instagram feed fail would be a critical problem for you._ (Of course, we're keenly aware that the legacy version has around 24 hours to live anyway! 😬).
> 
> Please check the [Project Wiki](https://github.com/stevenschobert/instafeed.js/wiki) for the migration guide and new options. For the quickest testing, grab the minified version of the js file here: https://github.com/stevenschobert/instafeed.js/blob/master/dist/instafeed.min.js or from jsDelivr: https://cdn.jsdelivr.net/gh/stevenschobert/instafeed.js@2.0.0rc1/src/instafeed.min.js
> 
> You can see [a super-simple demo of v2 at codepen](https://codepen.io/companionstudio/pen/rNVPGOz).

v2 condpe demo is not working

**Comment by soneji** [comment]
@benjamin-hull @stevenschobert any chance of this being released?

**Comment by stevenschobert** [comment]
Hey folks! I just officially dropped the RC tag and published [v2.0.0](https://github.com/stevenschobert/instafeed.js/releases/tag/v2.0.0) ([npm](https://www.npmjs.com/package/instafeed.js)).

The docs and unit tests are not up to where I want them yet, but there haven't been any bug reports with v2 in a while, and with my limited time I didn't want to keep the project in "release candidate" status forever.

I'll be closing out this issue — please do open new issues going forward if you find bugs with v2!
